app/js/controller/main_controller.js

```diff
@@ -16,10 +20,13 @@ export default class MainController extends Controller {
 	}
 
 	main() {
-		this.view.render();
-		window.addEventListener('hashchanged', this.route.bind(this));
-		this.route();
-		document.body.classList.remove('loading');
+		this.activityHandler.ready.then(route => {
+			window.location.hash = route;
```

Comment:
Note: I'm leaning towards the second alternative, removing the `route` logic from main controller. It seems simpler and more explicit.